### ISSUE TITLE
DeploymentTools.py: "from_size" may not be available

### ISSUE THREAD

**Issue by ArsThaumaturgis** [panda3d/panda3d#720]
When building a test-program via the "setuptools"-based deployment system, I hit the following error:
```text
Traceback (most recent call last):
  File "setup.py", line 95, in <module>
    'win_amd64'
  File "/usr/lib/python3/dist-packages/setuptools/__init__.py", line 129, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/dist/commands.py", line 1412, in run
    build_cmd.finalize_options()
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/dist/commands.py", line 373, in finalize_options
    iconobj.generateMissingImages()
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/p3d/DeploymentTools.py", line 351, in generateMissingImages
    if from_size > required_size:
UnboundLocalError: local variable 'from_size' referenced before assignment
```

Note that this seems to have come about as a result of attempting to build with a "setup.py" that specified icons that didn't exist (the "setup.py" file having come from a project that did have them). Building again with the icon-references removed resulted in the error no longer being emitted.

Still, this is a bit odd, as looking at the code the variable in question does seem to be established just above. o_0

**Comment by rdb** [comment]
Thanks.  This was already fixed in 28d9a33db44ead53b8a9ab06b5745531b1554f7e.

To fix this, make sure that you pass in at least one valid icon image (or none at all).

**Comment by ArsThaumaturgis** [comment]
Fair enough! I'm glad that it's fixed. ^_^

(And indeed, as you say, just removing the references to the icons allowed the build to proceed.)
